### DISCUSSION TITLE
Set DKIM private key for BYODKIM

### DISCUSSION THREAD

**Discussion by piotrblasiak** [aws/aws-sdk-java-v2#5126] [General]
How do you set the DKIM private key for when using BYODKIM instead of EasyDKIM?

**Comment by debora-ito** [comment]
@piotrblasiak can you give us more context about the question, and how you're using the SDK? Sharing a sample code would help.

**Reply by piotrblasiak** [reply]
In a previous version of the SDK, there was: 
https://javadoc.io/static/software.amazon.awssdk/sesv2/2.20.126/software/amazon/awssdk/services/sesv2/model/PutEmailIdentityDkimAttributesRequest.html

SES has the ability to DKIM sign messages, either by using its own keypairs (easydkim), or by supplying your own private key (BYODKIM). Here's the documentation explaining this: https://docs.aws.amazon.com/ses/latest/dg/send-email-authentication-dkim-bring-your-own.html

I want to be able to do this using the java sdk, but PutEmailIdentityDkimAttributesRequest is missing from the latest sdk (2.25.31).

**Reply by debora-ito** [reply]
> but PutEmailIdentityDkimAttributesRequest is missing from the latest sdk (2.25.31).

I can see PutEmailIdentityDkimAttributesRequest in the latest SDK version - https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/sesv2/model/PutEmailIdentityDkimAttributesRequest.html

The first link (2.20.126) is for a static javadoc.io url, maybe the reference is from a different place?

**Comment by piotrblasiak** [comment]
Ah now I see the problem - there is a ses and a sesv2 module (this class is only available in the latter)... That solves it. Thanks!
